main_logic/core.py

```
@@ -386,32 +395,43 @@ async def handle_text_data(self, text: str, is_first_chunk: bool = False):
 
         Only flushes TTS and sends turn_end to the frontend so that the
         realistic-queue buffer is flushed.  Does NOT trigger hot-swap,
         analyze_request, or agent-callback re-delivery — those belong
         exclusively to user-initiated conversation turns.
         """
-        if self.use_tts and self.tts_thread and self.tts_thread.is_alive():
+        delivered = bool(commit)
+        if delivered and self.use_tts and self.tts_thread and self.tts_thread.is_alive():
             try:
                 self.tts_request_queue.put((None, None))
             except Exception as e:
```

Ruff (0.15.9)

[warning] Do not catch blind exception: `Exception`

(BLE001)